### ISSUE TITLE
[bug report] expand the table row. If the table data is changed, the expanded row will be automatically stowed. Is there any way to prevent the table from being stowed

### ISSUE THREAD

**Issue by LadyChatterleyLover** [element-plus/element-plus#2913]
### Element Plus version
1.0.2-beta.70

### OS/Browsers version
none

### Vue version
3.2.2

### Reproduction Link
https://codepen.io/ladychatterleylover/pen/GRmzYxm

### Steps to reproduce
1. 展开行里面放入一个输入框,  v-model绑定props.row.name
2. 输入框每输入一次, 展开行会自动收起

### What is Expected?
期望输入内容展开行不收起

### What is actually happening?
是不是每次改变行的数据，表格的展开行会重新渲染？

**Comment by undefined-degisn** [comment]
表格会重新渲染，如果希望表格重新渲染后依然能保持下拉状态，或许你可以考虑
结合expand-change事件，记录下一个展开行的数组，传递给expand-row-keys属性，这样可以使得展开后重新渲染可以保持下拉状态

**Comment by angsuo** [comment]
> 表格会重新渲染，如果希望表格重新渲染后依然能保持下拉状态，或许你可以考虑
> 结合expand-change事件，记录下一个展开行的数组，传递给expand-row-keys属性，这样可以使得展开后重新渲染可以保持下拉状态

Yes, the expected-to-stay-expanded row will collapse (default state) on any data change.

If you want to keep it expanded:

1. Add an event handler to the `<el-table>`: `@expand-change="handleExpand"`.
2. Set `row-key="id"` to the `<el-table>`, where `id` should be a property name of the row.
3. Set `:expand-row-keys="expandedRowKeys"` to the `<el-table>`, where `expandedRowKeys` is an array of `id`s.

The `handleExpand` function will be called with 2 parameters: `(rowChanged, expandedRows)`, where `rowChanged` is the row which is collapsed or expanded, and `expandedRows` is an array of the expanded rows. 
Inside the `handleExpand` function `expandedRowKeys` should be updated according to the `expandedRows`'s value.

Then the expanded rows should keep the expanded state despite the data changes.

(Still a beginner, struggled a few hours to figure this out, hope this will help)

**Comment by 317607692** [comment]
如果单元格内放置的是图片，那么在展开行和取消展开行两种状态切换时，图片会闪烁

**Comment by LadyChatterleyLover** [comment]
这样感觉太麻烦了，有没有可能内部做一下优化，把这个功能加入进去




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "element-plus/element-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月22日(星期天) 下午5:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [element-plus/element-plus] [bug report] expand the table row. If the table data is changed, the expanded row will be automatically stowed. Is there any way to prevent the table from being stowed (#2913)





  
表格会重新渲染，如果希望表格重新渲染后依然能保持下拉状态，或许你可以考虑
 结合expand-change事件，记录下一个展开行的数组，传递给expand-row-keys属性，这样可以使得展开后重新渲染可以保持下拉状态
  
Yes, the expected-to-stay-expanded row will collapse (default state) on any data change.
 
If you want to keep it expanded:
  
Add an event handler to the <el-table&gt;: @expand-change="handleExpand".
 
Set row-key="id" to the <el-table&gt;, where id should be a property name of the row.
 
Set :expand-row-keys="expandedRowKeys" to the <el-table&gt;, where expandedRowKeys is an array of ids.
  
The handleExpand function will be called with 2 parameters: (rowChanged, expandedRows), where rowChanged is the row which is collapsed or expanded, and expandedRows is an array of the expanded rows.
 Inside the handleExpand function expandedRowKeys should be updated according to the expandedRows's value.
 
Then the expanded rows should keep the expanded state despite the data changes.
 
(Still a beginner, struggled a few hours to figure this out, hope this will help)
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by LadyChatterleyLover** [comment]
vue2版本是可以做到的，但vue3就不行了




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "element-plus/element-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月22日(星期天) 下午5:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [element-plus/element-plus] [bug report] expand the table row. If the table data is changed, the expanded row will be automatically stowed. Is there any way to prevent the table from being stowed (#2913)





  
表格会重新渲染，如果希望表格重新渲染后依然能保持下拉状态，或许你可以考虑
 结合expand-change事件，记录下一个展开行的数组，传递给expand-row-keys属性，这样可以使得展开后重新渲染可以保持下拉状态
  
Yes, the expected-to-stay-expanded row will collapse (default state) on any data change.
 
If you want to keep it expanded:
  
Add an event handler to the <el-table&gt;: @expand-change="handleExpand".
 
Set row-key="id" to the <el-table&gt;, where id should be a property name of the row.
 
Set :expand-row-keys="expandedRowKeys" to the <el-table&gt;, where expandedRowKeys is an array of ids.
  
The handleExpand function will be called with 2 parameters: (rowChanged, expandedRows), where rowChanged is the row which is collapsed or expanded, and expandedRows is an array of the expanded rows.
 Inside the handleExpand function expandedRowKeys should be updated according to the expandedRows's value.
 
Then the expanded rows should keep the expanded state despite the data changes.
 
(Still a beginner, struggled a few hours to figure this out, hope this will help)
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by LadyChatterleyLover** [comment]
好的




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "element-plus/element-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月13日(星期一) 晚上6:01
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [element-plus/element-plus] [bug report] expand the table row. If the table data is changed, the expanded row will be automatically stowed. Is there any way to prevent the table from being stowed (#2913)





 
如果单元格内放置的是图片，那么在展开行和取消展开行两种状态切换时，图片会闪烁
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Dsaquel** [comment]
I assume this issue has been fixed.
